homeassistant/components/eps/coordinator.py

```diff
@@ -0,0 +1,45 @@
+"""Coordinator for EPS integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyepsalarm import EPS
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, EPS_TO_HASS
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EPSDataUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Class to manage fetching EPS data."""
+
+    def __init__(self, hass: HomeAssistant, eps_api: EPS, site: str) -> None:
+        """Initialize global EPS data updater."""
+        self.eps_api = eps_api
+        self.state: str | None = None
+        self.site = site
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+
+    def _update_data(self) -> None:
+        """Fetch data from EPS via sync functions."""
+        status = self.eps_api.get_status()
+        _LOGGER.debug("EPS status: %s", status)
+        self.state = EPS_TO_HASS.get(status, status)
```

Comment:
I'd suggest to do this mapping in `alarm_control_panel.py` and let the coordinator be responsible for fetching the data.